app/server/lib/ExternalStorage.ts

```diff
@@ -226,13 +227,22 @@ export class ChecksummedExternalStorage implements ExternalStorage {
           const expectedChecksum = await this._options.sharedHash.load(fromKey);
           // Let null docMD5s pass.  Otherwise we get stuck if redis is cleared.
           // Otherwise, make sure what we've got matches what we expect to get.
-          // S3 is eventually consistent - if you overwrite an object in it, and then read from it,
-          // you may get an old version for some time.
+          // S3 is eventually consistent. However, times ago, if you overwrote an object in it,
```

Comment:
Instead of S3 is eventually consistent, I'd say:
AWS S3 was eventually consistent, but now has stronger guarantees:
  https://aws.amazon.com/blogs/aws/amazon-s3-update-strong-read-after-write-consistency/

"However, times ago, " -> "Previous to this change, "